src/components/TabComponent/TabComponent.js

```diff
@@ -0,0 +1,33 @@
+import React, {  PureComponent } from "react";
+import "./TabComponent.css";
+
+export default class TabComponent extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { activeTab: 0 };
+    //    this.setState(this.state);
+    this.onTabChange = (index) => {
+      this.setState({ activeTab: index });
+      this.props.tabChanged(this.state.activeTab);
```

Comment:
I want you to understand that setState may not be executed immediately so in this case _this.state.activeTab_ may contain the previous value and not the current one.
2 ways hot to avoid it: 

- create callback with tabChanged as the second argument of setState and then it will be call after new value will set in setState

```
const { tabChanged } = this.props;
const { activeTab } = this.state;

this.setState({ activeTab: index }, () => tabChanged(activeTab));
```

- call tabChanged with index, because it's index of current tab

```
const { tabChanged } = this.props;

this.setState({ activeTab: index });
tabChanged(index);
```